app/services/encrypted_document_storage/document_writer.rb

```diff
@@ -0,0 +1,35 @@
+module EncryptedDocumentStorage
+  class DocumentWriter
+    def encrypt_and_write_document(front_image:, back_image:)
+      key = SecureRandom.bytes(32)
+      encrypted_front_image = aes_cipher.encrypt(front_image, key)
+      encrypted_back_image = aes_cipher.encrypt(back_image, key)
+
+      front_image_reference = SecureRandom.uuid
+      back_image_reference = SecureRandom.uuid
+
+      storage.write_image(encrypted_image: encrypted_front_image, reference: front_image_reference)
+      storage.write_image(encrypted_image: encrypted_back_image, reference: back_image_reference)
+
+      WriteDocumentResult.new(
+        front_reference: front_image_reference,
+        back_reference: back_image_reference,
+        encryption_key: Base64.strict_encode64(key),
+      )
+    end
+
+    def storage
+      @storage ||= begin
+        if Rails.env.production?
```

Comment:
could we make this a config/feature flag instead of a rails env check? ex if a performance test environment or like a personal env doesn't need to have a bucket configured?

Reply:
There is going to be a feature flag to enable / disable this entirely. I just haven't gotten around to it yet.

Reply:
I do stand by the rails env check however. If a sandbox is going to be persisting the things that get uploaded and the sandboxes are on the public internet then requiring the tooling be in place to properly support encrypted image storage seems prudent. A feature flag will enable the feature to be disabled entirely in sandboxes.

Reply:
not 100% sure I follow but I think we're saying the same thing! My gut reaction was more of a "I think it's more 12-factor-like to have things be configurable than checking hardcoded envs like that" but same same

Reply:
Okay, I just added the feature flags. This code path is only reachable if `encrypted_document_storage_enabled` is true and it is false by default in production.

I can add a feature flag that allows us toggle the storage layer as well. My quiet concern is there aren't any situations where we want to turn on local storage in a deployed environment. On the other hand I can imagine a misconfiguration that leads to problems.

Lemme know what you think

Reply:
that seems good enough to me! 🚢